src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -506,19 +538,24 @@ public boolean validateKB(Model dataToBeValidated, Model union) {
         return validateAgainstShacl(unionModel, shacl);
     }
 
+    /**
+     * Returns the validation report from the last call to validateKB.
+     *
+     * @return the last validation report, or null if validateKB has not yet been called
+     */
+    public Resource getValidationReport() {
+        return validationReport;
+    }
+
     /**
      * Validates against the SHACL file to ensure that resources have the required properties
      * (and in some cases, only the required properties) of the proper types.  Returns true if
      * validation passes.
      */
     private boolean validateAgainstShacl(Model dataToBeValidated, Model shacl) {
```

Comment:
Hmm, as long as we clearly state in the JavaDocs that if it returns false/invalid, there's no way to find out what happened.  Now, I guess that's just part of the caveat of using this call-- only use this if you don't care about a reason...otherwise use the one that returns the validation report, or null if it's valid.  Also, we'll need to give them different names since the signature is the same, just with a different return value.  `validateKBAndReturnReport()` ?  Hmm, maybe it should return the report even if it's valid, but the caller will have to call `getRequiredProperty` on the report.  Or call it `validateKBAndReturnReportIfInvalid()`, which is a mouthful.  Thoughts?

I am open to making this change and updating the Examples and ScalingTest.  I do like removing the "state" from the validator.  I already had to update a couple VERDI files (which I haven't done yet) to reflect the changes to the validator-- expect a new VERDI PR later today once this PR is merged.  But AIDA-711 shouldn't be Done until the VERDI PR is also merged, don't you think?

Reply:
Made the changes we discussed so that the last report is not part of the validator's state.